File: chat/src/main/java/glitch/chat/object/json/RoomState.java
package glitch.chat.object.json;

import lombok.AllArgsConstructor;
import lombok.Data;

import java.util.Locale;

@Data
@AllArgsConstructor
public class RoomState {
    private Locale broadcasterLanguage;
    private boolean emoteOnly;
    private Long follow;
    private boolean r9k;
    private Long slow;
    private boolean subsOnly;

    public boolean isFollowersOnly() {
        return follow != -1L;
    }

    public boolean isSlowMode() {
        return slow > 0;
    }
}

File: chat/src/main/kotlin/glitch/chat/irc/RoomState.kt
package glitch.chat.irc

import glitch.api.objects.json.interfaces.IDObject
import java.util.*

/**
 *
 * @author Damian Staszewski [damian@stachuofficial.tv]
 * @version %I%, %G%
 * @since 1.0
 */
data class RoomState(
        override val id: Long,
        var broadcasterLanguage: Locale?,
        var isEmoteOnly: Boolean,
        var follow: Long,
        var isR9k: Boolean,
        var slow: Long,
        var isSubsOnly: Boolean
) : IDObject<Long> {
    val isFollowersOnly: Boolean
        get() = follow != -1L

    val isSlowMode: Boolean
        get() = slow > 0
}